index.d.ts

```diff
@@ -38,10 +38,10 @@ declare module 'robot3' {
    * @param state - The name of the destination state.
    * @param args - Any extra argument will be evaluated to check if they are one of Reducer, Guard or Action.
    */
-  export function transition<C>(
+  export function transition<C, E>(
```

Comment:
`E` is the event generic, right? We could infer the `type` string for this `event` object from that argument:

```ts
export type EventBase<Type extends string> = {
  type: Type;
  [key: string]: any;
};
export function transition<C, E extends EventBase<T>, T extends string>(
  event: T,
  state: string,
  ...args: (Reducer<C, E> | Guard<C, E> | Action<C, E>)[]
): Transition
```

Reply:
Yep, `E` is the event generic. The event can just be a string though in which case it wouldn't extend `EventBase`. I'm also not quite sure if this is worth it because the user now needs to provide three type arguments and duplicate the event type string here and in the function.